src/modules/iotjs_module_tls.c

```diff
@@ -210,32 +210,29 @@ JS_FUNCTION(TlsContext) {
   jerry_value_t jkey =
       iotjs_jval_get_property(joptions, IOTJS_MAGIC_STRING_KEY);
 
-  if (jerry_value_is_string(jcert) && jerry_value_is_string(jkey)) {
-    iotjs_string_t cert = iotjs_jval_as_string(jcert);
-    const char *cert_chars = iotjs_string_data(&cert);
+  iotjs_string_t bcert = iotjs_string_create();
+  iotjs_string_t bkey = iotjs_string_create();
```

Comment:
There is no need to call the `iotjs_string_create` functions here. The `iotjs_jval_get_data` will do the require calls to create the strings.

Reply:
We need it because if we don't get buffer or string as an argument, these two values remain uninitialized.

Reply:
That is not a problem, you should not use those variables if the conversion failed. The string destroy calls can be moved inside the `if` below.